public/app/features/trails/MetricSelect/api.ts

```diff
@@ -31,7 +34,9 @@ export async function getMetricNamesWithoutScopes(
   instances: string[],
   limit?: number
 ) {
-  const matchTerms = adhocFilters.map((filter) => `${filter.key}${filter.operator}"${filter.value}"`);
+  const matchTerms = adhocFilters.map((filter) =>
+    removeBrackets(queryModeller.renderLabels([{ label: filter.key, op: filter.operator, value: filter.value }]))
+  );
```

Comment:
Let's put this behind the feature flag too

Reply:
Good idea, done in https://github.com/grafana/grafana/pull/96067/commits/7cf6c3cd4cfaa09dc081bfeaace03dfd287957f4 ✅